release/test_candidate.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh -ex
+
+# https://cwiki.apache.org/confluence/display/COUCHDB/Testing+a+Source+Release
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+if test -n "$1"; then
+    version=$1
+else
+    echo "error: no version"
+    exit
+fi
+
+if test -n "$2"; then
+    candidate=$2
+else
+    echo "error: no candidate number"
+    exit
+fi
+
+log () {
+    printf "\033[1;31m$1\033[0m\n"
+}
+
+cd `dirname $0`
+
+basename=`basename $0`
+
+if test -n "$3"; then
+    tmp_dir=$3
+else
+    tmp_dir=`mktemp -d /tmp/$basename.XXXXXX` || exit 1
+    log "Creating temporary directory $tmp_dir"
+fi
+
+config_opts=${CONFIG_OPTS:-"--js-engine=quickjs --with-clouseau"}
+
+artifact_url=https://dist.apache.org/repos/dist/dev/couchdb/source/$version/rc$candidate/
+
+mkdir -p $tmp_dir/dist
+cd $tmp_dir/dist
+
+erl -eval 'erlang:display(erlang:system_info(otp_release)), halt().' -noshell
+elixir --version
+python --version
+java -version
+
+wget --no-parent --no-directories -r $artifact_url
```

Comment:
my version of this also pulls down the hashes and prints them next to the generated ones, so you can spot any difference right away.

Also slight preference to use `curl` as `wget` is non-default on BSDs and macOS.

Reply:
> my version of this also pulls down the hashes and prints them next to the generated ones, so you can spot any difference right away.

I'm happy to change it if you can show me an example of how your script does it.

Also, I tried to find a nice way to get `curl` to download an entire directory, but most things I read said that `wget` was the better tool for the job. Open to suggestions for the equivalent command though.

Reply:
For what it is worth, on FreeBSD [`fetch`](https://man.freebsd.org/cgi/man.cgi?fetch(1)) is the default downloader tool, that is part of the base system.  Both `curl` and `wget` could be installed as third-party packages, I am not sure if there is a generic preference for one or the other.

Reply:
I would prefer we do this outside of a release cycle, making structural changes like this could lead to delayed reviews on the official library, which we wanna avoid. 3.4.1 was a two week affair already